dmrpp_generator/main.py

```diff
@@ -90,29 +90,25 @@ def process(self):
         for granule in granules:
             dmrpp_files = []
             for file_ in granule['files']:
-                if not search(f"{self.processing_regex}$", file_['filename']):
-                    self.LOGGER_TO_CW.debug(f"{self.dmrpp_version}: regex {self.processing_regex} does not match filename {file_['filename']}")
+                if not search(f"{self.processing_regex}$", file_['fileName']):
+                    self.LOGGER_TO_CW.debug(f"{self.dmrpp_version}: regex {self.processing_regex} does not match filename {file_['fileName']}")
                     continue
-                self.LOGGER_TO_CW.debug(f"{self.dmrpp_version}: reges {self.processing_regex} matches filename to process {file_['filename']}")
-                output_file_paths = self.dmrpp_generate(input_file=file_['filename'],
-                                                       dmrpp_meta=dmrpp_meta)
+                self.LOGGER_TO_CW.debug(f"{self.dmrpp_version}: regex {self.processing_regex} matches filename to process {file_['fileName']}")
+                input_file_path = file_.get('filename', f's3://{file_["bucket"]}/{file_["key"]}')
```

Comment:
I'm not sure what the file object looks like here if you're generating from something local instead of in S3, so it still checks for the `filename` key and will use it if available, otherwise it will build up the S3 path as described in the Cumulus docs - https://nasa.github.io/cumulus/docs/upgrade-notes/update-task-file-schemas#updated-files-schema